src/wilson/dom.cljs

```diff
@@ -174,9 +175,19 @@
   (update elem 1 merge-attrs attrs))
 
 (defn with-class
-  "Adds a class to an element."
-  [cls elem]
-  (with-attrs {:class cls} elem))
+  "Adds a class to a component."
+  [cls component]
+  (let [parse-f2c (fn [render-fn & outer-props]
+                    (let [outer-comp (apply render-fn outer-props)]
+                      (fn [_ & inner-props]
+                        (let [inner-comp (apply outer-comp inner-props)]
+                          (with-attrs {:class cls} inner-comp)))))
+        [render-fn] (match component
+                      [render-fn & _]
+                      [render-fn])]
```

Comment:
Uh, right. 
It works on elements without attrs map already and tests are there (first `with-class-test` doesn't have attrs-map, same in `with-attrs-test`)